### PR TITLE
zstd: add version 1.5.1

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zstd/1.5.1**

1.5.1 is maintenance release with several performance improvements.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zstd/all/conanfile.py [hunk 1]
```diff
@@ -75,6 +78,7 @@ def package_info(self):
         self.cpp_info.components["zstdlib"].names["pkg_config"] = "libzstd"
         self.cpp_info.components["zstdlib"].names["cmake_find_package"] = zstd_cmake
         self.cpp_info.components["zstdlib"].names["cmake_find_package_multi"] = zstd_cmake
+        self.cpp_info.components["zstdlib"].set_property("cmake_target_name", f"{zstd_cmake}::{zstd_cmake}")
```
Comment (SpaceIm) [inline recipes/zstd/all/conanfile.py:81]: wrong target name

File: recipes/zstd/all/conanfile.py [hunk 2]
```diff
@@ -75,6 +78,7 @@ def package_info(self):
         self.cpp_info.components["zstdlib"].names["pkg_config"] = "libzstd"
```
Comment (SpaceIm) [inline recipes/zstd/all/conanfile.py:78]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "zstd")
        self.cpp_info.set_property("cmake_target_name", "zstd::{}".format(zstd_cmake))
        self.cpp_info.set_property("pkg_config_name", "libzstd")
        self.cpp_info.components["zstdlib"].set_property("pkg_config_name", "libzstd")
```

File: recipes/zstd/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [inline recipes/zstd/all/conanfile.py:81]:
```suggestion
        self.cpp_info.components["zstdlib"].set_property("cmake_target_name", "zstd::{}".format(zstd_cmake))
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Review by SSE4 — DISMISSED** [review]

File: recipes/zstd/all/conanfile.py [hunk 3]
```diff
@@ -58,6 +61,12 @@ def _patch_sources(self):
             tools.replace_in_file(os.path.join(self._source_subfolder, "build", "cmake", "lib", "CMakeLists.txt"),
                                   "POSITION_INDEPENDENT_CODE On", "")
 
+        # to avoid generating mismatched cputype objects from assembler codes (especially cross compiling)
+        if tools.Version(self.version) >= "1.5.1" and self.settings.arch != "x86_64":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "build", "cmake", "lib", "CMakeLists.txt"),
+                                  "file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c ${LIBRARY_DIR}/decompress/*.S)",
+                                  "file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c)")
+
```
Comment (SpaceIm) [inline recipes/zstd/all/conanfile.py:69]: A patch in upstream CMakeLists based on `CMAKE_SYSTEM_PROCESSOR` would be more elegant, and could be submitted upstream.

Comment (toge) [reply recipes/zstd/all/conanfile.py:69]: Thank you for your comments.
I am going to try to follow your idea.

Comment (toge) [reply recipes/zstd/all/conanfile.py]: Fixed.

**Review by SpaceIm — DISMISSED** [review]

File: recipes/zstd/all/patches/1.5.1-cmake-remove-asm-except-x86_64.patch [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+--- a/build/cmake/lib/CMakeLists.txt
++++ b/build/cmake/lib/CMakeLists.txt
+@@ -22,7 +22,11 @@ include_directories(${LIBRARY_DIR} ${LIBRARY_DIR}/common)
+ 
+ file(GLOB CommonSources ${LIBRARY_DIR}/common/*.c)
+ file(GLOB CompressSources ${LIBRARY_DIR}/compress/*.c)
+-file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c ${LIBRARY_DIR}/decompress/*.S)
++if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
++    file(GLOB DecompressSources ${LIBRARY_DIR}/decompress/*.c ${LIBRARY_DIR}/decompress/*.S)
++else(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
```
Comment (SpaceIm) [inline recipes/zstd/all/patches/1.5.1-cmake-remove-asm-except-x86_64.patch:12]: `else()` and `endif()` are fine also (less verbose & avoid repetition)

Comment (toge) [reply recipes/zstd/all/patches/1.5.1-cmake-remove-asm-except-x86_64.patch]: Thank you for your review.
I adopt your comment.

**Review by SpaceIm — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**